scripts/dropbox/migrate_to_external_accounts.py

```diff
@@ -0,0 +1,220 @@
+        oauth_key=user_settings_document['access_token'],
+        display_name=user_settings_document['dropbox_info'].get('display_name', None),
+    )
+    external_account.save()  # generate pk for external account
+    user.external_accounts.append(external_account)
+    user.save()
+    return external_account, user
+
+def make_new_user_settings(user):
+    # kill the old backrefs
+    database['user'].find_and_modify(
+        {'_id': user._id},
+        {
+            '$set': {
+                '__backrefs.addons.dropboxusersettings.owner': []
+            }
+        }
+    )
+    user.reload()
+    return user.get_or_add_addon('dropbox', override=True)
+
+def make_new_node_settings(node, node_settings_document, external_account=None, user_settings_instance=None):
+    # kill the old backrefs
+    database['node'].find_and_modify(
+        {'_id': node._id},
+        {
+            '$set': {
+                '__backrefs.addons.dropboxnodesettings.owner': []
+            }
+        }
+    )
+    node.reload()
+    node_settings_instance = node.get_or_add_addon('dropbox', auth=None, override=True)
+    node_settings_instance.folder = node_settings_document['folder']
+    node_settings_instance.save()
+    if external_account and user_settings_instance:
+        node_settings_instance.set_auth(
+            external_account,
+            user_settings_instance.owner,
+            log=False
+        )
+    return node_settings_instance
+
+def remove_old_documents(old_user_settings, old_user_settings_count, old_node_settings, old_node_settings_count, dry_run):
+    logger.info('Removing {0} old dropboxusersettings documents'.format(
+        old_user_settings_count
+    ))
+    if not dry_run:
+        database['dropboxusersettings'].remove({
+            '_id': {
+                '$in': map(lambda s: s['_id'], old_user_settings)
+            }
+        })
+    logger.info('Removing {0} old dropboxnodesettings documents'.format(
+        old_node_settings_count
+    ))
+    if not dry_run:
+        database['dropboxnodesettings'].remove({
+            '_id': {
+                '$in': map(lambda s: s['_id'], old_node_settings)
+            }
+        })
+
+def migrate(dry_run=True, remove_old=True):
+    user_settings_list = list(database['dropboxusersettings'].find())
+
+    # get in-memory versions of collections and collection sizes
+    user_settings_collection = database['dropboxusersettings']
+    old_user_settings = list(user_settings_collection.find())
+    old_user_settings_count = user_settings_collection.count()
+    node_settings_collection = database['dropboxnodesettings']
+    old_node_settings = list(node_settings_collection.find())
+    old_node_settings_count = node_settings_collection.count()
+
+    external_accounts_created = 0
+    migrated_user_settings = 0
+    migrated_node_settings = 0
+    for user_settings_document in user_settings_list:
+        verify_user_settings_document(user_settings_document)
```

Comment:
I'm getting an error when trying to run this script: `AssertionError: 'access_token' not found in {u'deleted': False, u'oauth_grants': {u'zujst': {u'565c7239c12510127240b91d': {}}}, u'_id': u'565c7239c12510127240b91e', u'_version': 1, u'owner': u'x7fn3'}`

The project `zujst` has revoked credentials, which explains why there's no access token.